cpp/src/transform/one_hot_encode.cu

```diff
@@ -117,7 +51,35 @@ std::pair<std::unique_ptr<column>, table_view> one_hot_encode(column_view const&
     return std::pair(std::move(empty_data), table_view{views});
   }
 
-  return type_dispatcher(input.type(), one_hot_encode_launcher{}, input, categories, stream, mr);
+  auto const total_size = input.size() * categories.size();
+  auto all_encodings =
+    make_numeric_column(data_type{type_id::BOOL8}, total_size, mask_state::UNALLOCATED, stream, mr);
+
+  auto const t_lhs = table_view{{input}};
+  auto const t_rhs = table_view{{categories}};
+  auto const comparator =
+    cudf::experimental::row::equality::two_table_comparator{t_lhs, t_rhs, stream};
+  auto const d_equal =
+    comparator.equal_to(nullate::DYNAMIC{has_nested_nulls(t_lhs) || has_nested_nulls(t_rhs)});
+
+  thrust::transform(
+    rmm::exec_policy(stream),
+    thrust::make_counting_iterator(0),
+    thrust::make_counting_iterator(total_size),
+    all_encodings->mutable_view().begin<bool>(),
+    [input_size = input.size(), d_equal] __device__(size_type i) {
+      auto const element_index  = cudf::experimental::row::lhs_index_type{i % input_size};
+      auto const category_index = cudf::experimental::row::rhs_index_type{i / input_size};
+      return d_equal(element_index, category_index);
+    });
+
+  auto const split_iter =
+    make_counting_transform_iterator(1, [width = input.size()](auto i) { return i * width; });
+  std::vector<size_type> split_indices(split_iter, split_iter + categories.size() - 1);
+
+  auto const encodings_view = table_view{split(all_encodings->view(), split_indices, stream)};
```

Comment:
This cannot be `const` because we are going to `move` it. So everything else needs to be `const` except the one that we are going  to `move` or `return`.
Sorry for the confusion. 
```suggestion
  auto encodings_view = table_view{split(all_encodings->view(), split_indices, stream)};
```

Reply:
Yes sorry, I should have picked that up myself - my attention is a bit divided at the moment